app/hooks/useMaxHeight.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import useMobile from "./useMobile";
+import useWindowSize from "./useWindowSize";
+
+const useMaxHeight = ({
+  elementRef,
+  maxViewportHeight = 90,
```

Comment:
I don't love this naming, might be better to pass as a fraction or… `maxViewportPercentage` for clarity